### DISCUSSION TITLE
pydanticModel.model_dump_json() to use serializer mode

### DISCUSSION THREAD

**Discussion by 97k** [567-labs/instructor#928] [General]
I just realised even if we use `Field(exclude=True)`, it doesn't get excluded from the json dump until we specify the mode.

For example

```python
from pydantic import BaseModel, Field

class Model(BaseModel):
    name: str
    allowed_names: str = Field(default=['name1', 'name2'], exclude=True)

print('Model dump - Usual')
pp.pprint(Model.model_json_schema())

print("Model dump serializer mode")
pp.pprint(Model.model_json_schema(mode="serialization"))

```
The output of this code is 

![Screenshot 2024-08-16 at 8 51 19 PM](https://github.com/user-attachments/assets/33a586aa-ca5f-4b22-b889-f847e12d65cc)

I don't want whatever I mark it as exclude to be send it to LLM.
Is it possible to do so?
@jxnl / Team

**Comment by jxnl** [comment]
In a flight right now but search json skip in our docs on python.useinstructor.com

**Reply by 97k** [reply]
Thanks for this
https://python.useinstructor.com/concepts/fields/?h=skip#omitting-fields-from-schema-sent-to-the-language-model

Btw, A thought, we can also use `basemodelclass.model_dump_json(mode='serialization')` in instructor's `generate_response.py` 🤔 
as if someone explicitly mentioned, exclude = True, then we should not send it to LLM?
